jax/_src/scipy/stats/_core.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2022 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import namedtuple
+from functools import partial
+from typing import Tuple
+
+import jax.numpy as jnp
+import scipy
+from jax import jit
+from jax._src.api import vmap
+from jax._src.numpy.lax_numpy import _check_arraylike
+from jax._src.numpy.util import _wraps
+from jax._src.util import canonicalize_axis
+
+ModeResult = namedtuple('ModeResult', ('mode', 'count'))
```

Comment:
Let's use
```python
from typing import NamedTuple
from jax._src.typing import Array
class ModeResult(NamedTuple):
  mode: Array
  count: Array
```

Reply:
```NamedTuple``` as the ```count``` method, so I'll use ```counts``` instead. And also ```modes``` instead of ```mode```.

Reply:
I think we should keep it as `mode` and `count` to match the numpy API. Those names don't seem to cause issues when Itry them

Reply:
The problem occurs when using mypy

Reply:
OK, in that case let's keep the current approach

Reply:
Correct runtime API is more important than type annotations